app/src/main/java/com/codingblocks/cbonlineapp/course/CourseViewModel.kt

```diff
@@ -221,4 +226,56 @@ class CourseViewModel(
         }
         return LivePagedListBuilder<String, Course>(dataSourceFactory, config)
     }
+
+    fun changeWishlistStatus(id: String){
+        runIO {
+            when (val response = repo.checkIfWishlisted(id)) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    if (isSuccessful) {
+                        if (response.value.body()?.id!=null){
+                            response.value.body()?.let { removeWishlist(it.id) }
+                        }else{
+                            addWishlist(id)
+                        }
+                    } else {
+                        setError(fetchError(code()))
+                    }
+                }
+            }
+        }
+    }
+
+    fun addWishlist(id: String){
+        val wishlist = Wishlist(Course(id))
```

Comment:
did you check this?

Reply:
Yes, I checked that and I removed the User id which we were sending before but we need to send the course id, in the parameters.